request.go

```diff
@@ -0,0 +1,88 @@
+package cfdns
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type request[T any] struct {
+	client      *Client
+	method      string
+	path        string
+	queryParams url.Values
+	headers     http.Header
+	body        *T
```

Comment:
I think we can use any. Initially it had a more specific type constraint, but I changed to `any`. I'll leave this for last, but I think I'll probably change it.

I can't think on a good reason to ever use `[T any]` instead of just `any` as a parameter to a function :).

Reply:
Changed to any. Thanks for the suggestion.